### PR TITLE
chore: update dependency grpc to 1.24.2

### PR DESCRIPTION
**Pull request by mayurkale22**

### CODE REVIEW

**Comment by mayurkale22** [comment]
This is based on https://github.com/census-instrumentation/opencensus-node/issues/715, @draffensperger and @lovrop please review

File: packages/opencensus-instrumentation-grpc/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
   "dependencies": {
     "@opencensus/core": "^0.0.19",
     "@opencensus/propagation-binaryformat": "^0.0.19",
-    "grpc": "~1.21.0",
+    "grpc": "1.24.2",
```
Comment (lovrop) [inline packages/opencensus-instrumentation-grpc/package.json]: `^1.24.2` would be more flexible and IMO fit better due to the nature of the dependency (per the _Additional context_ section on #715). Did you intentionally choose an exact version match?

Comment (mayurkale22) [reply packages/opencensus-instrumentation-grpc/package.json]: done in https://github.com/census-instrumentation/opencensus-node/pull/750/commits/a78335a7c7a542eb8f368034a061c7f6ab3f4eb7

**Comment by lovrop** [comment]
Looks good to me, thanks for the tweak. Seems @draffensperger needs to sign off?

**Comment by mayurkale22** [comment]
@draffensperger Please review when you get a chance.

File: packages/opencensus-instrumentation-grpc/test/test-grpc.ts [hunk 2]
```diff
@@ -546,7 +546,7 @@ describe('GrpcPlugin() ', function() {
                 grpcModule.status.OK
               );
               if (method.method === grpcClient.unaryMethod) {
-                assertStats(testExporter, 170);
+                assertStats(testExporter, 188);
```
Comment (draffensperger) [inline packages/opencensus-instrumentation-grpc/test/test-grpc.ts]: Out of curiosity, why are the numbers changing here? Will that confuse users at all?

Comment (mayurkale22) [reply packages/opencensus-instrumentation-grpc/test/test-grpc.ts]: The latest gRPC release (`grpc@1.24.2`) added new options (`flags`) to Metadata class due to this size of bytes send increased a little bit (we measure the sent_bytes_per_rpc).

Proposal: https://github.com/grpc/proposal/blob/master/L48-node-metadata-options.md
PR: https://github.com/grpc/grpc-node/pull/796

**Pull request merged**